src/main/java/com/fasterxml/jackson/core/util/BufferRecyclerProvider.java

```diff
@@ -0,0 +1,19 @@
+package com.fasterxml.jackson.core.util;
+
+import com.fasterxml.jackson.core.TokenStreamFactory;
+
+/**
+ * Provider for {@link BufferRecycler}s to allow pluggable and optional
+ * recycling of underlying input/output buffers.
+ *
+ * @since 2.16
+ */
+public abstract class BufferRecyclerProvider
```

Comment:
Is there any reason why this is an abstract class instead of an interface?

Reply:
I like abstract classes, but interface probably makes more sense here. Will change.